src/main/java/uk/gov/hmcts/reform/sendletter/services/date/DateCalculator.java

```diff
@@ -1,28 +1,61 @@
 package uk.gov.hmcts.reform.sendletter.services.date;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sendletter.services.date.holidays.BankHolidaysClient;
 
 import java.time.DayOfWeek;
+import java.time.LocalDate;
 import java.time.ZonedDateTime;
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static java.util.stream.Collectors.toList;
 
 @Service
 public class DateCalculator {
 
+    private static final Logger log = LoggerFactory.getLogger(DateCalculator.class);
+
+    private final BankHolidaysClient bankHolidaysClient;
+
+    public DateCalculator(BankHolidaysClient bankHolidaysClient) {
+        this.bankHolidaysClient = bankHolidaysClient;
+    }
+
     public ZonedDateTime subtractBusinessDays(ZonedDateTime dateTime, int numberOfBusinessDays) {
         int daysSubtracted = 0;
         ZonedDateTime adjustedDateTime = dateTime;
 
+        List<LocalDate> bankHolidays = getBankHolidayDates();
+
         while (daysSubtracted < numberOfBusinessDays) {
             adjustedDateTime = adjustedDateTime.minusDays(1);
-            if (isBusinessDay(adjustedDateTime)) {
+            if (isBusinessDay(adjustedDateTime, bankHolidays)) {
                 daysSubtracted++;
             }
         }
 
         return adjustedDateTime;
     }
 
-    private boolean isBusinessDay(ZonedDateTime dateTime) {
-        return dateTime.getDayOfWeek().getValue() < DayOfWeek.SATURDAY.getValue();
+    private boolean isBusinessDay(ZonedDateTime dateTime, List<LocalDate> bankHolidays) {
+        return dateTime.getDayOfWeek().getValue() < DayOfWeek.SATURDAY.getValue()
+            && !bankHolidays.contains(dateTime.toLocalDate());
+    }
+
+    private List<LocalDate> getBankHolidayDates() {
+        try {
+            return this.bankHolidaysClient
```

Comment:
Not a part of this pull request, but maybe the client could be specific to England (it doesn't support any other option) and then it would have a method called `getHolidays` instead of `getForEngland`? This would simplify it. For example it could have the whole URL to the holidays endpoint in configuration.

Reply:
done